dhall-lsp-server/README.md

```diff
@@ -1,9 +1,75 @@
-# dhall-lsp-server
+# Dhall language support in VSCode/ium
 
-This is a [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) server implementation for the [Dhall](https://dhall-lang.org) programming language.
+The Dhall language integration consists of the following parts:
+- The VSCode/ium plugin "Dhall Language Support" *([vscode-language-dhall](https://github.com/PanAeon/vscode-language-dhall))* adds syntax highlighting for Dhall files.
+- The VSCode/ium plugin "Dhall LSP Server" *([vscode-dhall-lsp-server](https://github.com/PanAeon/vscode-dhall-lsp-server))* implements the LSP client &ndash; yes, there is a naming issue here &ndash; that communicates with the backend via the [LSP protocol](https://microsoft.github.io/language-server-protocol/specification) to provide advanced language features like error diagnostics or type information, etc.
+- [*dhall-lsp-server*](https://github.com/dhall-lang/dhall-haskell/tree/master/dhall-lsp-server), which is part of the [*dhall-haskell*](https://github.com/dhall-lang/dhall-haskell) project, implements the actual LSP server (i.e. the backend) that implements those language features in an editor agnostic way, though at the moment only a VSCode/ium frontend exists.
 
+# Installation
 
-For installation or development instructions, see:
+The "official" releases can be installed as follows:
 
-* [`dhall-haskell` - `README`](https://github.com/dhall-lang/dhall-haskell/blob/master/README.md)
+- **vscode-language-dhall** should be installed directly from VSCode/ium via the extensions marketplace.
+- **vscode-dhall-lsp-server** can also be installed directly from the marketplace.
+- **dhall-lsp-server** can be installed from hackage with `cabal install dhall-lsp-server`.
 
+## Installing the latest development versions
+
+**Note&nbsp;** The versions of *vscode-dhall-lsp-server* and *dhall-lsp-server* need not necessarily match: an older client version will simply not expose all commands available in the backend, while an older server might not implement all commands exposed in the UI.
+
+**vscode-dhall-lsp-server**
+1. You need to have *npm* installed (e.g. using your favourite package manager).
+2. Clone `git@github.com:PanAeon/vscode-dhall-lsp-server.git` into a folder under `~/.vscode-oss/extensions/` (or `~/.vscode/extensions/` if you VSCode rather than VSCodium).
```

Comment:
I suggest giving people exact command lines to run for these instructions, such as:

```bash
$ git clone git@github.com:PanAeon/vscode-dhall-lsp-server.git ~/.vscode/extensions/vscode-dhall-lsp-server
```

Reply:
Good point!